.github/workflows/ci.yml

```diff
@@ -319,3 +319,26 @@ jobs:
     steps:
       - run: |
           tr -d '\n' <<< '${{ toJSON(needs.*.result) }}' | grep -q -v -E '(failure|cancelled)'
+
+  notify-tests-completed-failures:
+    if: ${{ always() && needs.tests-completed.result == 'failure' && (github.head_ref == github.ref_name == 'main' || startsWith(github.ref_name, 'release/')) }}
```

Comment:
`github.head_ref == github.ref_name == 'main'` is this intentional?

Reply:
Good catch! It's not. For context, `head_ref` gives you the branch name for PRs, whereas `ref_name` gives the branch name for non-PRs. I messed this up when switching between testing and having this ready to go.

There, strangely, isn't a single property that gives you the branch name universally, so I had to do it this way.